[BUG] vmiLatency measurement fails for delete jobs

# Bug Description
When running the example kubevirt-density workload, the vmiLatency measurement will throw an error for idle-job or delete-job, but works fine for the create job type.

## **Output of `kube-burner` version**
Version: 1.10.0
Git Commit: 1b9c7446fff5605724f0d8f72c38d855112e9b2a
Build Date: 2024-06-24T16:11:46Z
Go Version: go1.21.11
OS/Arch: linux amd64

## **Describe the bug**
 There is no error when the podLatency measurement is used for the `jobType: delete` kubevirt-density jobs

These jobs do disable the measurement collection:
`VMI latency measurement not compatible with delete jobs, skipping" file="vmi_latency.go:283`

However StopWatcher hits issues when the vmiLatency measurement is defined for the "delete" type.

## **To Reproduce**

Run the example kubevirt-density workload, ex: 
OBJ_REPLICAS=10  KUBECONFIG=/root/kubeconfig-test1 /root/kubevirt/kube-burner/kube-burner init  -c kubevirt-density.yml

## **Expected behavior**
vmiLatency does not error for delete

## **Screenshots or output**

Error messages:

```
time="2024-06-28 14:30:23" level=info msg="Job idle-job took 1m0s" file="job.go:196"
time="2024-06-28 14:30:23" level=info msg="Stopping measurement: vmiLatency" file="factory.go:125"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1a948a6]

goroutine 75 [running]:
github.com/kube-burner/kube-burner/pkg/measurements/metrics.(*Watcher).StopWatcher(0x0)
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/metrics/watcher.go:69 +0x86
github.com/kube-burner/kube-burner/pkg/measurements.(*vmiLatency).stop.func1()
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/vmi_latency.go:386 +0x25
github.com/kube-burner/kube-burner/pkg/measurements.(*vmiLatency).stop(0x3627360?)
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/vmi_latency.go:391 +0x82
github.com/kube-burner/kube-burner/pkg/measurements.Stop()
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/factory.go:126 +0x148
github.com/kube-burner/kube-burner/pkg/burner.Run.func1()
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:199 +0x1853
created by github.com/kube-burner/kube-burner/pkg/burner.Run in goroutine 1
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:101 +0x634

[...]

time="2024-06-28 14:27:17" level=info msg="Job delete-job took 1m14s" file="job.go:196"
time="2024-06-28 14:27:17" level=info msg="Stopping measurement: vmiLatency" file="factory.go:125"
panic: close of closed channel

goroutine 10 [running]:
github.com/kube-burner/kube-burner/pkg/measurements/metrics.(*Watcher).StopWatcher(0xc0007a1110)
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/metrics/watcher.go:73 +0xfe
github.com/kube-burner/kube-burner/pkg/measurements.(*vmiLatency).stop.func1()
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/vmi_latency.go:386 +0x25
github.com/kube-burner/kube-burner/pkg/measurements.(*vmiLatency).stop(0x3627360?)
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/vmi_latency.go:391 +0x82
github.com/kube-burner/kube-burner/pkg/measurements.Stop()
	/home/runner/work/kube-burner/kube-burner/pkg/measurements/factory.go:126 +0x148
github.com/kube-burner/kube-burner/pkg/burner.Run.func1()
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:199 +0x1853
created by github.com/kube-burner/kube-burner/pkg/burner.Run in goroutine 1
	/home/runner/work/kube-burner/kube-burner/pkg/burner/job.go:101 +0x634`

```

--- a/pkg/measurements/vmi_latency.go
+++ b/pkg/measurements/vmi_latency.go
@@ -279,13 +279,6 @@ func (p *vmiLatency) start(measurementWg *sync.WaitGroup) error {
 	defer measurementWg.Done()
 	// Reset latency slices, required in multi-job benchmarks
 	p.latencyQuantiles, p.normLatencies = nil, nil
-	if factory.jobConfig.JobType == config.DeletionJob {
-		log.Info("VMI latency measurement not compatible with delete jobs, skipping")
-		return nil
-	}
-	if err := p.validateConfig(); err != nil {
-		return err
-	}
 	p.metrics = make(map[string]*vmiMetric)
 	log.Infof("Creating VM latency watcher for %s", factory.jobConfig.Name)
 	restClient := newRESTClientWithRegisteredKubevirtResource()

